flux-desugar/src/desugar.rs

```diff
@@ -346,6 +345,13 @@ impl<'a, 'tcx> DesugarCtxt<'a, 'tcx> {
                 fhir::Ty::Array(Box::new(ty), fhir::ArrayLen)
             }
             surface::TyKind::Slice(ty) => fhir::Ty::Slice(Box::new(self.desugar_ty(None, *ty)?)),
+            surface::TyKind::Tuple(tys) => {
+                fhir::Ty::Tuple(
+                    tys.into_iter()
+                        .map(|ty| self.desugar_ty(None, ty))
+                        .collect::<Result<_, _>>()?,
```

Comment:
you can use `try_collect` from `itertools` (or `try_collect_exhaust` if we want to report all the errors)

Reply:
done! (but see above!)